sklearn/svm/src/libsvm/svm.cpp

```diff
@@ -442,6 +446,20 @@ double Kernel::dot(const PREFIX(node) *px, const PREFIX(node) *py)
 }
 #endif
 
+void Kernel::set_blas(BlasFunctions *blas)
+{
+	if (Kernel::blas_status==0)
+		Kernel::m_blas=blas;
+	Kernel::blas_status++;
```

Comment:
is blas_status really necessary ?
I thought 
```
if (!Kernel::m_blas)
    Kernel::m_blas=blas;
```
would be sufficient

Reply:
Thanks for your advice and good to hear from you.
This is needed actually since `Kernel::m_blas` is a static member for `class Kernel`. If you abandon the use of `blas_status` as a reference counter, you will get  an segmentation fault when you run `predict` after `fit`.
```
      model = ssvm.SVC(**self.build_opts)
      model.fit(self.data_split[0], self.data_split[1])
      predictions = model.predict(self.data[1])
```
This is for the reason that if you call `fit` at first time, the cython wrapper will pass an scipy blas function `dot` to `Kernel::m_blas`. However, when you call `predict` later, it will never fall into the condition you propose and won't assign `Kernel::m_blas` since it still points to a certain memory address. But the scipy blas function which is supposed to at the memory address is freed by the cython wrapper, which will leads to a segmentation fault.

Reply:
This does not seem to be threadsafe. For example, running this with the threading backend with `joblib` will break with this PR:

```py
from sklearn.svm import SVC
from sklearn.preprocessing import scale
from sklearn.datasets import make_classification
from joblib import Parallel, delayed

X, y = make_classification(random_state=0)
X = scale(X)


@delayed
def fit_and_score():
    est = SVC(kernel="linear").fit(X, y)
    return est.score(X, y)


scores = Parallel(n_jobs=4,
                  backend='threading')(fit_and_score() for i in range(500))
print(scores)
```